### PR TITLE
Remove warning triangle and center error message for video select errors

### LINKED ISSUE #1103: The warning triangle in case of a technical problem when sharing content seems lonely
@oas777 wrote in https://github.com/elan-ev/opencast-studio/pull/1069#issuecomment-1675100147:

> The warning triangle in case of a technical problem when sharing content seems lonely: Could we integrate it into the (upper right corner) of the warning text?

to which @LukasKalbertodt replied in https://github.com/elan-ev/opencast-studio/pull/1069#issuecomment-1686631451:

> Imagine the warning triangle being a placeholder for the video what would have appeared there. That was the idea anyway.
What do others think about this? Moving this would be easy.

Poor warning triangle. 😦

### PR DESCRIPTION
**Pull request by LukasKalbertodt**
Fixes #1103

This also fixes this error case for narrow screens where the error message previously overflowed the container, sometimes even the viewport.

Test deployment: https://test.studio.opencast.org/2023-09-26_12-46-54-pr1113-6312834670/

### CODE REVIEW

**Comment by lkiesow** [comment]
I think the (missing) padding around the warning box looks pretty weird.

Screenshot from 2023-09-15 13-16-50


Simply adding more padding makes this look a lot better:
Screenshot from 2023-09-15 13-17-35

If you really want it this small, then make the whole box red:

Screenshot from 2023-09-15 13-16-50

**Comment by LukasKalbertodt** [comment]
Thanks for comment, I indeed completely missed the weird horizontal margin on wider screens. I agree that looks weird. I removed that now so that the margin around the red box is always the same on all sides. Your suggestion of adding lots more margin is difficult to implement due to the auto-sizing algorithm that's involved here. And that's especially tricky with two video inputs.

I personally prefer having the white box around it still. I think the contrast of the red background to the grey background is not too great. What do you think? Please keep in mind that this is a rather rare situation where legibility is the most important aspect I'd say. Most normal users will never see this error. 

Also CC @narickmann @oas777 you commented on the issue, but let's continue the discussion here.

image

image

image

**Comment by oas777** [comment]
I'm usually a friend of padding/canvasses, but here I would prefer to remove the white box. But I don't have strong feelings regarding this...

**Comment by oas777** [comment]
PS: Revisiting this a couple of hours later, my feelings are getting stronger for removing the white box. @narickmann?

**Comment by LukasKalbertodt** [comment]
I removed the margin now. With also the box shadow removed in those cases, i would agree that it looks better. [Test for yourself](https://test.studio.opencast.org/2023-09-27_10-00-18-pr1113-6324369015/).

image
image

**Comment by oas777** [comment]
I think it does, obviously... but from my high uniformity horse I cannot oversee the fact that the two boxes are different in height... 

grafik

PS: Why is the text different with my settings?

**Comment by LukasKalbertodt** [comment]
@oas777  The boxes contain different texts, so that's why their height is different. One could make them a fixed height but that's not ideal, especially since in different languages the text could take up less or more space. Also, there is one error message (when you stop a running screen share for example) that is very short and would look weird in a large box. Again remember that this is something people will very rarely see.

Not sure what your PS question is. Surely you don't mean that you have German text because your Studio was set to German?

---

@lkiesow are your concerns addressed by the latest push?

**Comment by oas777** [comment]
From a uniformity perspective, the length of the text is relevant when complaining about the height of the boxes, but maybe I should stop before asking for uniform error messages with the same text lengths...

**Comment by oas777** [comment]
PS: My error message is "Cannot access display", the one you shared starts with "Video stream unexpectedly...", both are for content, I presume. What did you do to trigger yours?

**Comment by LukasKalbertodt** [comment]
> What did you do to trigger yours?

I stopped an active screen share via my browser.

---

It sounds like this could be merged then. Will just ask someone to review the code.

**Comment by oas777** [comment]
> > What did you do to trigger yours?
> 
> I stopped an active screen share via my browser.
>

I was able to reproduce this, so... 
> It sounds like this could be merged then. Will just ask someone to review the code.

... yes.

File: src/steps/video-setup/preview.tsx [hunk 1]
```diff
@@ -86,7 +93,15 @@ const PreviewVideo: React.FC<{ input: Input }> = ({ input }) => {
...
     let inner: JSX.Element;
     if (allowed === false || unexpectedEnd) {
-      inner = <FiAlertTriangle css={{ fontSize: 48, color: COLORS.danger4 }} />;
+      inner = <div>
+        {allowed === false && <ErrorBox
+          css={{ margin: 0 }}
+          title={t(`source-${input.isDesktop ? "display" : "user"}-not-allowed-title`)}
+          body={t(`source-${input.isDesktop ? "display" : "user"}-not-allowed-text`)}
+        />}
+        {/* TODO: differentiate between desktop and camera for better error */}
```
Comment (Arnei) [inline src/steps/video-setup/preview.tsx:102]: What about this ToDo?

**Review by Arnei — CHANGES_REQUESTED** [review]
Code looks reasonable.

There is still a warning triangle if you cause an error during recording (e.g. by revoking permissions). Is that intended?

**Comment by LukasKalbertodt** [comment]
> There is still a warning triangle if you cause an error during recording (e.g. by revoking permissions). Is that intended?

~~I had hoped nobody would notice that~~ Good point, thanks! I fixed that now. Now unexpected stream end in the recording step looks like in the video select step.

But your comment also made me notice another bug that I also fixed. Should be good to go now.

File: src/steps/video-setup/preview.tsx (hunk 1 above)
Comment (LukasKalbertodt) [reply src/steps/video-setup/preview.tsx:102]: That was there before and I don't think this PR necessarily needs to fix that.

**Pull request merged**